### PR TITLE
Update MMGIS API vector layer functions

### PR DESCRIPTION
**Pull request by ac-61**
This removes the `keepN` parameter from the `mmgisAPI.updateVectorLayer` and creates 2 helper functions for removing existing features on vector layers. Splitting the trim functionality into separate functions solves the issue of whether or not we should include new input data when removing features.

The `trimVectorLayer` function removes features on a specified layer before a specified time: 
`window.mmgisAPI.trimVectorLayer(layerName, keepAfterTime, timePropPath)`

The `keepLastN` function retains the same functionality as the original `keepN` parameter in the `mmgisAPI.updateVectorLayer` function (i.e.  removes features on a specified layer to keep the specified number of existing features):
`window.mmgisAPI.keepLastN(layerName, keepLastN)`

The `keepLastN` function is unaware of time so it will remove features based on the order the features were added to the specified vector layer.

### CODE REVIEW

File: docs/pages/markdowns/JavaScript_API.md [hunk 1]
```diff
@@ -52,6 +52,38 @@ window.mmgisAPI.updateVectorLayer('Waypoints', {
...
+```
+
+### keepLastN(layerName, keepLastN)
+
+This function removes features on a specified layer to keep the specified number of existing features. This function is not aware of time and will only keep the previous N number of features based on the order the features were added to the layer.
+
+#### Function parameters
+
+- `layerName` - name of layer to update
+- `timePropPath` - name of time property to compare with the time specified by keepAfterTime
```
Comment (tariqksoliman) [inline docs/pages/markdowns/JavaScript_API.md:78]: `timePropPath` should be removed here.

Comment (ac-61) [reply docs/pages/markdowns/JavaScript_API.md]: Good catch! I updated in 10df53099c555f93ef25cfe13610f3f9c22fc6da.

**Review by tariqksoliman — APPROVED** [review]
LGTM!

File: docs/pages/markdowns/JavaScript_API.md [hunk 2]
```diff
@@ -52,6 +52,37 @@ window.mmgisAPI.updateVectorLayer('Waypoints', {
...
+#### Function parameters
+
+- `layerName` - name of layer to update
+- `keepAfterTime` - absolute time in the format of YYYY-MM-DDThh:mm:ssZ; will keep all features after this time
+- `timePropPath` - name of time property to compare with the time specified by keepAfterTime
+
+The following is an example of how to call the `trimVectorLayer` function:
+
+```javascript
+window.mmgisAPI.trimVectorLayer('Waypoints',  "2021-12-01T15:10:00.000Z", 'time')
```
Comment (jtroberts) [inline docs/pages/markdowns/JavaScript_API.md:68]: The addition of milliseconds is slightly different than the format string above, but I think JavaScript handles it all the same anyway.

**Review by jtroberts — APPROVED** [review]

**Comment by ac-61** [comment]
@tariqksoliman @jtroberts I updated the pull request to add additional functions. `window.mmgisAPI.trimVectorLayer` is now `window.mmgisAPI. trimVectorLayerKeepAfterTime` and has a matching function to trim in the opposite direction `window.mmgisAPI. trimVectorLayerKeepBeforeTime`.  `window.mmgisAPI.keepLastN` remains the same and now has an extra friend, `window.mmgisAPI.keepFirstN`.

**Pull request merged**